integreat_cms/cms/models/pages/page.py

```diff
@@ -197,6 +197,16 @@ class Page(AbstractTreeNode, AbstractBasePage):
         verbose_name=_("Ignore HIX value"),
         help_text=_("This allows to ignore HIX value for this specific page."),
     )
+    embedded_offer = models.ForeignKey(
+        "cms.OfferTemplate",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
```

Comment:
```suggestion
        "cms.Offer",
        null=True,
        blank=True,
        on_delete=models.PROTECT,
```
I realize it's quite late for this debate, but one idea I just had: We could also convert this to a foreign key to the `Offer` model, then we'd have the protection mechanism in the region form by default...

Reply:
Haha oh I'm so sorry, somehow I missed that we refactored our offer models [two years ago](https://github.com/digitalfabrik/integreat-cms/pull/790) :joy:
We used to have an intermediary [`Offer` model](https://github.com/digitalfabrik/integreat-cms/blob/5425956ffa8a1c0c9c36953e86c57f7593c86352/src/cms/models/offers/offer.py#L10-L31) that acted as link between regions and offer templates^^
Ok well then my idea doesn't work of course, since protection only protects against deletion of the templates, not the removal of the regions :facepalm:

Reply:
Oh, alright 😄 (Funnily enough I constantly find myself writing `Offer` or `OfferProvider`, seem I'm still influenced by the Compass)